src/main/java/com/github/angleshq/angles/listeners/testng/AnglesTestngTestListener.java

```diff
@@ -0,0 +1,40 @@
+package com.github.angleshq.angles.listeners.testng;
+
+import com.github.angleshq.angles.exceptions.AnglesPropertyNotGivenException;
+import com.github.angleshq.angles.listeners.AbstractAnglesListener;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+public class AnglesTestngTestListener extends AbstractAnglesListener implements ITestListener {
+
+    public AnglesTestngTestListener() throws AnglesPropertyNotGivenException {
+        initialiseAnglesTestParameters();
+    }
+
+    public void onStart(ITestContext context) {
+        startAnglesBuild();
+    }
+
+    public void onFinish(ITestContext context) {
+
+    }
+
+    public void onTestStart(ITestResult result) {
+        startAnglesTest(result.getMethod().getRealClass().getSimpleName(), result.getMethod().getMethodName());
```

Comment:
Unfortunately this is called after the @BeforeMethod (which means the reporter doesn't capture that in the results).